cachix/src/Cachix/Client/Commands.hs

```diff
@@ -224,11 +227,21 @@ getPushParams env pushOpts name = do
   pushSecret <- findPushSecret (config env) name
   store <- wait (storeAsync env)
   authToken <- Config.getAuthTokenMaybe (config env)
-  return $
-    PushParams
-      { pushParamsName = name,
-        pushParamsSecret = pushSecret,
-        pushParamsClientEnv = clientenv env,
-        pushParamsStrategy = pushStrategy store authToken pushOpts name,
-        pushParamsStore = store
-      }
+  let token = fromMaybe (Token "") authToken
```

Comment:
Hmm, why this code was changed? AFAICS it doesn't really affect any compression setup

Reply:
I need to provide `compression` field of the `BinaryCache` to the pushStrategy creation to select, which of two compression methods use. Domen showed me this method of getting binary cache.

Reply:
oh, now I see what it was needed